pacman/utilities/utility_objs/resource_tracker.py

```diff
@@ -1261,8 +1262,13 @@ def get_maximum_resources_available(self, chips=None, processor_id=None,
                 best_processor_id = self._best_core_available(chip, key,
                                                               processor_id)
                 processor = chip.get_processor_with_id(best_processor_id)
-                max_dtcm_available = processor.dtcm_available
-                max_cpu_available = processor.cpu_cycles_available
+                try:
+                    max_dtcm_available = processor.dtcm_available
+                    max_cpu_available = processor.cpu_cycles_available
+                except AttributeError:
+                    if processor is None:
+                        raise PacmanProcessorNotAvailableError(
+                            chip_x, chip_y, best_processor_id)
```

Comment:
After talking, this just needs a re-raise to ensure the original exception is raised if it isn't caused by None

Reply:
How embarrassing!

I had assume I had done that as it is required.

Fixed.